src/IpcEventDispatcher.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#include "IpcEventDispatcher.h"
+#include "IpcChannel.h"
+#include "PointViewerMainWindow.h"
+
+
+IpcEventDispatcher::IpcEventDispatcher(QObject* parent, PointViewerMainWindow* getPayload, 
+                                       QByteArray hookSpec, QByteArray hookPayload, int whichShortCut)
+    : QObject(parent),
```

Comment:
The hooks are registered and managed in HookManager, and the message corresponding to a hook event is formatted in HookFormatter, the dispatching being taken care of by simply connecting these objects with the appropriate channel via signals and slots. I'll change the names.